.gitignore

```diff
@@ -22,3 +22,5 @@
 npm-debug.log*
 yarn-debug.log*
 yarn-error.log*
+
+yarn.lock
```

Comment:
So, on the backend we decided to let the dependencies lock file in GIT, which is a common pattern in order to make sure that a given version is tight to given dependencies.

Should we do the same here or not? Or is this role devoted to just package.lock and not yarn.lock?

Reply:
Shouldn't package.json already have the exact versions? In a different context, I found checking in the .lock files to be more confusing than helping.
Of course that requires exact versions in package.json and not a "catch all" version > 0.0

Reply:
package.json may or may not have the exact version. Having the .lock file in git is like a proof of any dependency update.

Reply:
See also "Tilde ranges" and "Caret ranges" that are frequently used : https://github.com/npm/node-semver#tilde-ranges-123-12-1

Reply:
Well with a tilde range we say "update at will" and then encode the result in the lock file.
Why not pick a specific version on purpose and encode this in the package.json?

Reply:
I imagine they are good reasons (and others not so good) for that most javascripters use this instead of exact versions. For instance I believe any security related patches are automatically picked up that way.

Also, I wonder if transitive dependencies that are selected with tilde ranges upstream can be updated here even if the upstream project didn't change. I'm not sure if I'm clear, here is an example:
sws depends on foo "=1.0.0"
foo depends on bar "~1.0.0"
We build sws, we get foo 1.0.0 and bar 1.0.0

Then bar is updated to 1.0.1
We rebuild sws without changing any dependency, but even if we point strictly to foo 1.0.0, we get bar 1.0.1
This change could only be seen in the lock file I guess.

Reply:
@jotak @pilhuhn  I'm fine with locking the versions down. It should be a different jira though

Reply:
It is best practice to check-in the package-lock.json file to ensure consistent package dependencies.
https://stackoverflow.com/questions/44206782/do-i-commit-the-package-lock-json-file-created-by-npm-5